src/ray/gcs/gcs_server/gcs_object_manager.cc

```diff
@@ -226,6 +226,33 @@ void GcsObjectManager::RemoveObjectLocationInCache(const ObjectID &object_id,
   }
 }
 
+void GcsObjectManager::RemoveAllObjectLocationsInCacheByJobId(const JobID &job_id) {
+  absl::MutexLock lock(&mutex_);
+
+  std::unordered_set<ObjectID> matched_object_ids;
+  for (auto &kv : object_to_locations_) {
+    if (kv.first.TaskId().JobId() == job_id) {
+      matched_object_ids.insert(kv.first);
+    }
+  }
+
+  for (auto &key : matched_object_ids) {
+    object_to_locations_.erase(key);
+  }
+
+  std::unordered_set<ClientID> idle_nodes;
+  for (auto &kv : node_to_objects_) {
+    for (auto &value : matched_object_ids) {
+      kv.second.erase(value);
+    }
+    if (kv.second.empty()) idle_nodes.insert(kv.first);
+  }
+
+  for (auto &node : idle_nodes) {
```

Comment:
Can you comment the purpose of this code block?

Reply:
no problem :)